File: integer-part.cpp
#include <iostream>
using namespace std;

int integerPart(double x) {
    int p;
    if (x > 0) {
        p = 0;
        while (x > 0) {
            p = p + 1;
            x = x - 1;
        }
    } else {
        //for negative numbers, the integer part, 
        // is the largest integer less than or equal to x.
        p = -1; 
        while (x < 0) {
            p = p - 1;
            x = x + 1;
        }
    }
    return p;
}

int main() {
    double x;
    cout << "Enter a real number: ";
    cin >> x;

    int p = integerPart(x);
    cout << "The integer part is: " << p << endl;

    return 0;
}

File: sesion1/exercises/integerPart.py
'''
Propose a method based on addition, subtraction, and comparison
operations for determining the integer part of a real number. The integer part (sometimes
called the floor and denoted for a real number x as ⌊x⌋) of a real number is defined as the
largest integer that is less than or equal to the given real number. For example, the integer
part of 2.4 is 2, and for -2.4, it is -3.
'''

f = 2.4
i = 0


if f > 0:
    while i < f-1:
        i += 1
else:
    while i > f:
        i -= 1

print(i) 